### PR TITLE
Fix python formatting everywhere

### PR DESCRIPTION
**Pull request by npentrel**

### CODE REVIEW

File: docs/components/board/_index.md [hunk 1]
```diff
@@ -448,10 +448,11 @@ Get an [`DigitalInterrupt`](#digital_interrupts) by `name.`
 For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/viam/components/board/index.html#viam.components.board.Board.digital_interrupt_by_name).
 
 ```python
-my_board = Board.from_robot(robot=robot, name=)
+my_board = Board.from_robot(robot=robot, name="my_board")
```
Comment (andf-viam) [inline docs/components/board/_index.md:451]: Nice catch! No way `flake8` caught this one???

File: docs/components/camera/_index.md [hunk 2]
```diff
@@ -133,7 +133,7 @@ frame = await my_cam.get_image()
 
 # Convert "frame" to a standard 2D image representation.
 # Remove the 1st 3x8 bytes and reshape the raw bytes to List[List[Int]].
-standard_frame frame.bytes_to_depth_array()
+standard_frame = frame.bytes_to_depth_array()
```
Comment (andf-viam) [inline docs/components/camera/_index.md:136]: This too! Surely this is `naomi8` ??

File: docs/tutorials/get-started/confetti-bot.md [hunk 3]
```diff
@@ -294,10 +295,10 @@ Copy this code and add it to your own code within the main function block:
 ```python {class="line-numbers linkable-line-numbers"}
 while True:
     print(party_return_value.get())
-    while (await party_return_value.get()) == True:
+    while (await party_return_value.get()):
         await start.set_power(.8)
         await asyncio.sleep(0.1)
-        if (await GPIO.get()) == False:
+        if not (await GPIO.get()):
```
Comment (andf-viam) [inline docs/tutorials/get-started/confetti-bot.md:301]: Oh I like this tool!

File: docs/tutorials/get-started/confetti-bot.md [hunk 4]
```diff
@@ -294,10 +295,10 @@ Copy this code and add it to your own code within the main function block:
 ```python {class="line-numbers linkable-line-numbers"}
 while True:
     print(party_return_value.get())
-    while (await party_return_value.get()) == True:
+    while (await party_return_value.get()):
```
Comment (andf-viam) [inline docs/tutorials/get-started/confetti-bot.md:298]: 😂

File: docs/tutorials/services/color-detection-scuttle.md [hunk 5]
```diff
@@ -315,81 +319,86 @@ You could also write some code with a Viam SDK to [make your rover move in a squ
...
+
+
 async def connect():
-   creds = Credentials(
+    creds = Credentials(
         type="robot-location-secret",
-        payload="[PLEASE ADD YOUR SECRET HERE. YOU CAN FIND THIS ON THE CODE SAMPLE TAB]")
-   opts = RobotClient.Options(
+        payload="[PLEASE ADD YOUR SECRET HERE. "
+                "YOU CAN FIND THIS ON THE CODE SAMPLE TAB]")
```
Comment (andf-viam) [inline docs/tutorials/services/color-detection-scuttle.md:335]: I wonder if we should exclude this -- users will think the creds are two parts (like our [org API keys](https://docs.viam.com/manage/cli/#create-an-organization-api-key) are, for example, easy to confuse here!!). 

I would vote for a single line,  but shorter text:

```suggestion
        payload="<location_secret>")
```

Or `payload='YOUR LOCATION SECRET'  # The location secret`
or `location-secret`,
or ...

Like on many other code samples. Same with the robot address just below.

**Review by andf-viam — APPROVED** [review]
Wow this is amazing!! LGTM with just one concern about multi-line credential placeholders.

Fantastic job on this, this will help so much!!

File: docs/components/board/_index.md (hunk 1 above)
Comment (npentrel) [reply docs/components/board/_index.md:451]: it did tell me there was an error - I did manually have to fix it though

File: docs/components/camera/_index.md (hunk 2 above)
Comment (npentrel) [reply docs/components/camera/_index.md:136]: flake8 says error - naomi fixes

File: docs/tutorials/get-started/confetti-bot.md (hunk 4 above)
Comment (npentrel) [reply docs/tutorials/get-started/confetti-bot.md:298]: the word is `pythonic`

File: README.md [hunk 6]
```diff
@@ -43,6 +43,12 @@ python3 -m http.server 9000 --directory public
 
 ## Test the docs locally
 
+To ensure all python snippets are properly formatted before creating a PR, install [flake8-markdown](https://github.com/johnfraney/flake8-markdown) and add the following like to `.git/hooks/pre-commit`:
+
+```sh
+flake8-markdown "docs/**/*.md"
```
Comment (npentrel) [inline README.md:49]:
```suggestion
flake8-markdown  $(git diff --diff-filter=d --name-only HEAD | grep '\.md$')
```

File: README.md [hunk 7]
```diff
@@ -43,6 +43,12 @@ python3 -m http.server 9000 --directory public
 
 ## Test the docs locally
 
+To ensure all python snippets are properly formatted before creating a PR, install [flake8-markdown](https://github.com/johnfraney/flake8-markdown) and add the following like to `.git/hooks/pre-commit`:
```
Comment (npentrel) [inline README.md:46]:
```suggestion
To ensure all python snippets are properly formatted before creating a commit, install [flake8-markdown](https://github.com/johnfraney/flake8-markdown) and add the following like to `.git/hooks/pre-commit`:
```

File: docs/components/gantry/_index.md [hunk 8]
```diff
@@ -363,7 +366,8 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/_modules
 ```python
 my_gantry = Gantry.from_robot(robot=robot, name="my_gantry")
 
-# Stop all motion of the gantry. It is assumed that the gantry stops immediately.
+# Stop all motion of the gantry. It is assumed that the gantry stops
+# immediately.
```
Comment (JessamyT) [inline docs/components/gantry/_index.md:370]: Is there a way to make the two lines more evenly distributed automatically? Here and some other places, there are just one or two words on the second line and I wish it'd automatically do something like 
```suggestion
# Stop all motion of the gantry. It is assumed that the
# gantry stops immediately.
```

Comment (npentrel) [reply docs/components/gantry/_index.md:370]: I don't mind accepting this change but that's more personal preference, the only linting rule is for lines not to exceed 80 characters. Just probably won't go looking for other instances right now.

**Pull request merged**